### PR TITLE
windows: 64bit support

### PR DESCRIPTION
**Pull request by nillerusr**
- ~Saverestore is broken~
- ~Release build crashing~
- ~Explosion particles not displaying~ ( only under cdb debugger )
- Protocol is broken
- Maybe something else

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -47,6 +41,17 @@ jobs:
         ./waf.bat configure -T debug
         ./waf.bat build
 
+  build-windows-amd64:
+    runs-on: windows-2019
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build windows-i386
```
Comment (LegendaryGuard) [inline .github/workflows/build.yml:49]: Be careful! The name is wrong, it isn't i386, it's amd64.

Comment (nillerusr) [reply .github/workflows/build.yml:49]: shit happens

**Comment by EnderZip** [comment]
try launching the game without debugger attached or -dev
that kinda fixed the particles on my side

**Comment by nillerusr** [comment]
@EnderZip 
> game without debugger attached
Lol that's works.

**Comment by EnderZip** [comment]
Yeah strange ain't it, btw is sound working?

**Comment by nillerusr** [comment]
@EnderZip yeap is working

**Comment by EnderZip** [comment]
Ah well good to hear, keep up the good work

**Comment by nillerusr** [comment]
Well, protocol is broken

**Comment by EnderZip** [comment]
So you can't load into a map yet?

**Comment by LegendaryGuard** [comment]
The maps can be loaded successfully, but there are stuff to fix when I used `.\waf.bat configure -T release --64bits` to obtain the release build: 
- When the projectile from Rocket Propelled Grenade is shot, the projectile touches a wall or something and doesn't collide and explode as should. The projectile stays stuck in the wall/ground and moves very slowly depending where you're guiding.

image


- Also SMG grenades, Crossbow arrow gets stuck incorrectly in the wall and pheropods doesn't mark the collision, just stays.

image

image

image



- Story mode: In Entanglement chapter, Alyx, after seeing and talking her dad, in the door doesn't open it and doesn't move from her place.

image


- In the console, this message is spamming, it happens when the player is coming near/touching a wall or something does:
```bash
AllocTouchLink: failed to allocate touchlink_t.
```

In the debug build (`.\waf.bat configure -T debug --64bits -8`), appears an error after Assertion window (https://github.com/nillerusr/source-engine/issues/131#issue-1422741781) and the game doesn't load: 

CTSListBaseERROR

**Comment by nillerusr** [comment]
@LegendaryGuard that's strange. I don't have problems with touchlink_t

**Comment by nillerusr** [comment]
@LegendaryGuard Don't use saves from 32bit engine

**Comment by LegendaryGuard** [comment]
> @LegendaryGuard Don't use saves from 32bit engine

If that's so, let me try without these saves when I can. 👌

**Comment by LegendaryGuard** [comment]
> @LegendaryGuard Don't use saves from 32bit engine

Worked. 👍 

> In the debug build (`.\waf.bat configure -T debug --64bits -8`), appears an error after Assertion window (https://github.com/nillerusr/source-engine/issues/131#issue-1422741781) and the game doesn't load: 
CTSListBaseERROR

Still I can't use the debug build, I tried attaching the process with VS IDE, I can't obtain any debug info... I think we won't be able to debug. 🤔 


> Well, protocol is broken

If it's broken, that means no multiplayer games can be played. 🤔

**Comment by nillerusr** [comment]
@LegendaryGuard 

> no multiplayer games can be played

yep, i even can't connect to local server

**Comment by LegendaryGuard** [comment]
I've a noob dev question: 
How I compile to get HL2 Deathmatch or CS: Source? I never tried to compile specifically for a game like these (also HL1, TF2, DoD, Portal, ...), the target is always HL2 (I always get the same compiled stuff with this game directory: `hl2/bin`).

**Comment by nillerusr** [comment]
@LegendaryGuard 
`./waf.bat --help`

**Comment by LegendaryGuard** [comment]
Is normal for the compiled launcher to appear with the same name and icon: `hl2_launcher`?
image

EDIT: Moved to the last comment.

**Comment by maotouxie** [comment]
Server uses different class tables

**Comment by nillerusr** [comment]
@LegendaryGuard of course that's normal, lol.
Run using `-game hl2mp` command-line argument

**Comment by nillerusr** [comment]
@HappyDOGE bullshit, у меня развалился протокол

**Comment by nillerusr** [comment]
@LegendaryGuard Can you check cstrike on 64 bit?

**Pull request closed without merging**